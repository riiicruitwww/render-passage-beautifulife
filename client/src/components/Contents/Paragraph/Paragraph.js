import React from 'react';
import PropTypes from 'prop-types';
import ChunkContainer from 'containers/ChunkContainer';

const Paragraph = ({ paragraph }) => {
  const renderParagraph = () => {
    return paragraph.children.map((item, index) => (
      <ChunkContainer key={index} chunk={item} />
    ));
  };

  return (
    <p className="Paragraph" style={paragraph.style}>
      {renderParagraph()}
    </p>
  );
};

Paragraph.propTypes = {
  paragraph: PropTypes.shape({
    children: PropTypes.array.isRequired
  }).isRequired
};

export default Paragraph;
