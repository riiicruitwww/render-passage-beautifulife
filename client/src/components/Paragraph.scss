.Paragraph {
  margin-bottom: 1em;
}
