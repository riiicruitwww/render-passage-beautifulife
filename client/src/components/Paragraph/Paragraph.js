import React from 'react';
import ChunkContainer from '../../containers/ChunkContainer';

const Paragraph = ({ paragraph }) => {
  const renderParagraph = () => {
    return paragraph.children.map((item, index) => (
      <ChunkContainer key={index} chunk={item} />
    ));
  };

  return (
    <p className="Paragraph" style={paragraph.style}>
      {renderParagraph()}
    </p>
  );
};

export default Paragraph;
