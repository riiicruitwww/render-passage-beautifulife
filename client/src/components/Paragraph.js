import React from 'react';
import './Paragraph.scss';
import { renderPropertyByName } from './renderPropertyByName';

const Paragraph = ({ paragraph }) => {
  const renderParagraph = () => {
    return paragraph.children.map((item, index) => {
      return renderPropertyByName(item, index);
    });
  };

  return (
    <p className="Paragraph" style={paragraph.style}>
      {renderParagraph()}
    </p>
  );
};

export default Paragraph;
