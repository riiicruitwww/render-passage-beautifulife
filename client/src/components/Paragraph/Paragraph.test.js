import React from 'react';
import { shallow } from 'enzyme';
import Paragraph from './Paragraph';

describe('Paragraph', () => {
  describe('about render', () => {
    const props = {
      paragraph: {
        children: []
      }
    };
    const wrapper = shallow(<Paragraph {...props} />);

    it('compare with snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('component rendered', () => {
      expect(wrapper.find('.Paragraph').length).toBe(1);
    });
  });
});
