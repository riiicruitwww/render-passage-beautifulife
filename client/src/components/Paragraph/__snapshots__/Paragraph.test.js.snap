// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph about render compare with snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Paragraph
    paragraph={
      Object {
        "children": Array [],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "className": "Paragraph",
      "style": undefined,
    },
    "ref": null,
    "rendered": Array [],
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "Paragraph",
        "style": undefined,
      },
      "ref": null,
      "rendered": Array [],
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
